### PR TITLE
[BEAM-9481] Exclude signature files from expansion service test dependencies

### PR DESCRIPTION
**Pull request by TheNeuralBit**
Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python2_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python35_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)
XLang | --- | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

File: sdks/java/testing/expansion-service/build.gradle [hunk 1]
```diff
@@ -49,7 +49,7 @@ task buildTestExpansionServiceJar(type: ShadowJar) {
     )
   }
   exclude 'META-INF/*.RSA', 'META-INF/*.SF','META-INF/*.DSA'
-  from { project.configurations.testRuntime.collect { it.isDirectory() ? it : zipTree(it) }}
+  from { project.configurations.testRuntime.collect { it.isDirectory() ? it : zipTree(it) }} { exclude 'META-INF/*.RSA', 'META-INF/*.SF', 'META-INF/*.DSA' }
```
Comment (lukecwik) [inline sdks/java/testing/expansion-service/build.gradle]:
```suggestion
configurations = [
            project.configurations.testRuntime
          ]
```

File: sdks/java/testing/expansion-service/build.gradle [hunk 2]
```diff
@@ -49,7 +49,7 @@ task buildTestExpansionServiceJar(type: ShadowJar) {
     )
   }
   exclude 'META-INF/*.RSA', 'META-INF/*.SF','META-INF/*.DSA'
```
Comment (lukecwik) [inline sdks/java/testing/expansion-service/build.gradle]:
```suggestion
          exclude "META-INF/INDEX.LIST"
          exclude "META-INF/*.SF"
          exclude "META-INF/*.DSA"
          exclude "META-INF/*.RSA"
```

**Comment by lukecwik** [comment]
JIRA?

Note that this is a problem in SQL as well for the shadowJar and the testShadowJar:
https://github.com/apache/beam/blob/1988284a89b10b60eea48325f8a3b370b551c77c/sdks/java/extensions/sql/jdbc/build.gradle#L56

**Comment by TheNeuralBit** [comment]
> Note that this is a problem in SQL as well for the shadowJar and the testShadowJar:
I was under the impression that SQL was ok because it uses `shadowJar` directly. But I'll admit I'm not well versed in how all this works. I'll look into whether we're pulling in signatures in the sql jars.

**Comment by lukecwik** [comment]
You

> > Note that this is a problem in SQL as well for the shadowJar and the testShadowJar:
> 
> I was under the impression that SQL was ok because it uses `shadowJar` directly. But I'll admit I'm not well versed in how all this works. I'll look into whether we're pulling in signatures in the sql jars.

Even if you may not be pulling them in right now, you may start pulling them in the future with a dependency change.

Also, there are other flags that should be set like zip64. Take a look at the shadowJar and shadowTestJar tasks in BeamModulePlugin.groovy

**Comment by Ardagan** [comment]
I created jira for this
https://issues.apache.org/jira/browse/BEAM-8919

**Comment by Ardagan** [comment]
I suggest to merge this PR now to mitigate issue and add further improvements later.

**Pull request merged**